builtin/credential/cert/path_certs.go

```diff
@@ -147,6 +169,47 @@ func (b *backend) pathCertWrite(
 	policies := policyutil.ParsePolicies(d.Get("policies"))
 	allowedNames := d.Get("allowed_names").([]string)
 
+	// Parse the ttl (or lease duration)
+	systemDefaultTTL := b.System().DefaultLeaseTTL()
+	ttl := time.Duration(d.Get("ttl").(int)) * time.Second
+	if ttl == 0 {
+		ttl = time.Second * time.Duration(d.Get("lease").(int))
+	}
+	if ttl > systemDefaultTTL {
+		return logical.ErrorResponse(fmt.Sprintf("Given ttl of %d seconds is greater than current mount/system default of %d seconds", ttl/time.Second, systemDefaultTTL/time.Second)), nil
+	}
+
+	// Parse max_ttl
+	systemMaxTTL := b.System().MaxLeaseTTL()
+	maxTTL := time.Duration(d.Get("max_ttl").(int)) * time.Second
+	if maxTTL > systemMaxTTL {
+		return logical.ErrorResponse(fmt.Sprintf("Given max_ttl of %d seconds is greater than current mount/system default of %d seconds", maxTTL/time.Second, systemMaxTTL/time.Second)), nil
+	}
+
+	// Logic pertaining to interaction between ttl and max_ttl
+	if maxTTL != 0 {
+		// If ttl is 0, set the ttl as the max ttl
+		if ttl == 0 {
+			ttl = maxTTL
```

Comment:
Don't do this; this will happen automatically in core if TTL is zero and there is a max TTL.